src/ApplicationInsights.ServiceFabric.Native/Net45/RequestTrackingModule/RequestTrackingUtils.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.ApplicationInsights.DataContracts;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace Microsoft.ApplicationInsights.ServiceFabric.Module
+{
+    /// <summary>
+    /// Util methods to share some implemetation across service remoting V1 and V2.
+    /// </summary>
+    internal static class RequestTrackingUtils
+    {
+        private static DataContractSerializer _baggageSerializer = new DataContractSerializer(typeof(IEnumerable<KeyValuePair<string, string>>));
+
+        /// <summary>
+        /// Updates the given telemetry object with the provided methodName, parentId and property bag
+        /// </summary>
+        /// <param name="rt">Telemetry object.</param>
+        /// <param name="methodName">method name for the request telemetry.</param>
+        /// <param name="parentId">Correlation parent Id.</param>
+        /// <param name="correlationBytes">Correlation property bag as byte array.</param>
+        public static void UpdateTelemetryBasedOnCorrelationContext(RequestTelemetry rt, string methodName, string parentId, byte[] correlationBytes)
+        {
+            if (!string.IsNullOrEmpty(parentId))
+            {
+                rt.Context.Operation.ParentId = parentId;
+                rt.Context.Operation.Id = GetOperationId(parentId);
+            }
+
+            rt.Name = methodName;
+
+            if (correlationBytes != null)
+            {
+                var baggageBytesStream = new MemoryStream(correlationBytes, writable: false);
+                var dictionaryReader = XmlDictionaryReader.CreateBinaryReader(baggageBytesStream, XmlDictionaryReaderQuotas.Max);
+                var baggage = _baggageSerializer.ReadObject(dictionaryReader) as IEnumerable<KeyValuePair<string, string>>;
+                foreach (KeyValuePair<string, string> pair in baggage)
+                {
+                    rt.Context.Properties.Add(pair.Key, pair.Value);
```

Comment:
please also check if there is no such key in the properties and do not overwrite if the key is present.